Sources/XCMetricsPlugins/GitPlugin.swift

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2021 Spotify AB.
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import Foundation
+import XCMetricsClient
+import XCMetricsUtils
+
+public struct GitPlugin {
+    
+    public enum GitData {
+        case branch // the current branch name
+        case latestSHA // the most recent commit hash in short format (6)
```

Comment:
I think we can add a few more fields here if you want (also possible to do in another PR if you prefer):
- `is_dirty`: `git status --short` if empty or not.
- `branch_creation_date`: `git reflog show --date=iso main` and parse the date.
- `user_email`: `git config --get user.email` and hash the email with MD5 using CryptoSwift (it's already in the project, just need to add a dependency in Package.swift). This could be useful to report problems if someone gives us their git email and wants us to look into issues, but optional since we already have other ways to find logs in case a developer chooses to gives us their hashed username.

Reply:
I just added `isDirty` and `userEmail(redacted:)` options on 58c9a910821dcf673778e0ca9d43ee91ce1aba95 which will be enabled by default.  I didn't however, have time to get to the `branch_creation_date` field due to the date parsing and branch name considerations and my availability right now.  I'm happy to follow up in another PR in the future with those (and maybe more fields?) if that's OK!

Reply:
That's fine, thanks for the contribution so far!